app/assets/javascripts/pageflow/editor/models/file_uploader.js

```diff
@@ -15,13 +17,23 @@ pageflow.FileUploader = pageflow.Object.extend({
       fileType: fileType
     });
 
+    if (editor.nextUploadTargetFile){
+      file.set({parent_file_id: editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: editor.nextUploadTargetFile.fileType().typeName});
+    }
+
```

Comment:
What happens when a top level file upload is started (via drag and drop) while a `nextUploadTargetFile` is set?